<launch>

    <!-- default args -->
    <arg name="verbose" default="false" />
    <arg name="frame_child" default="base_link" />
    <arg name="vehicle_urdf" default="cat $(find vehicle_core)/data/nessie.urdf" />

    <!-- use Nessie AUV throttle limit -->
    <param name="thrusters/throttle_limit" value="85"/>

    <!-- load the simulation config -->
    <rosparam command="load" file="$(find vehicle_core)/conf/simulation_nav.yaml"/>
    <rosparam command="load" file="$(find vehicle_core)/conf/simulation_model.yaml"/>
    <rosparam command="load" file="$(find vehicle_core)/conf/thruster_model.yaml"/>

    <param name="robot_description" command="$(arg vehicle_urdf)"/>
    
    <!-- start the simulated thrusters -->
    <node name="thrusters_simulator" pkg="vehicle_core" type="node_thruster_sim.py" output="screen">
        <!-- set topics input and outputs -->
        <param name="topic_input" value="thrusters/commands"/>
        <param name="topic_feedback" value="thrusters/status"/>
        <param name="topic_forces" value="forces/sim/body"/>
    </node>

    <!-- start the navigation simulator -->
    <node name="nav_sim" pkg="vehicle_core" type="node_nav_sim.py" output="screen">
        <!-- sim_rate: publish rate of nav messages (default: 20 Hz) -->
        <param name="sim_rate" value="20"/>
        <param name="pub_rate" value="10"/>

        <param name="verbose" value="$(arg verbose)" type="bool"/>

        <!-- odometry configuration (defaults are used for single vehicle operations) -->
        <param name="topic_odom" value="nav/odometry"/>
        <param name="frame_parent" value="world"/>
        <param name="frame_odom" value="map"/>
        <param name="frame_child" value="$(arg frame_child)"/>
    </node>

</launch>
